lib/postgrex/replication.ex

```diff
@@ -836,6 +690,23 @@ defmodule Postgrex.Replication do
           :ok -> {:noreply, s}
           {error, reason, protocol} -> reconnect_or_stop(error, reason, protocol, s)
         end
+
+      {:query, query, mod_state} ->
+        case s do
+          %{repl_opts: nil, copy_opts: nil} ->
+            case Protocol.handle_simple(query, [], s.protocol) do
+              {:ok, %Postgrex.Result{} = result, protocol} ->
+                handle(mod, :handle_result, [result, mod_state], %{s | protocol: protocol}, from)
+
+              {error, reason, protocol} ->
+                from && reply(from, {error, reason})
+                reconnect_or_stop(error, reason, protocol, %{s | state: {mod, mod_state}})
```

Comment:
I like this option. @sorentwo, you will probably need to consider something similar. Maybe `handle_result` receives `{:ok, _} | {:error, Postgrex}`. Although it may also return `:disconnect` and then we must disconnect.